app/controllers/review_mapping_controller.rb

```diff
@@ -562,49 +487,110 @@ def response_report
-      @avg_and_ranges= @assignment.compute_avg_and_ranges_hash
-    when "FeedbackResponseMap"
-      #Example query
-      #SELECT distinct reviewer_id FROM response_maps where type = 'FeedbackResponseMap' and 
-      #reviewed_object_id in (select id from responses where 
-      #map_id in (select id from response_maps where reviewed_object_id = 722 and type = 'ReviewResponseMap'))
-      @review_response_map_ids = ResponseMap.select("id").where(["reviewed_object_id = ? and type = ?", @id, 'ReviewResponseMap'])
-      @response_ids = Response.select("id").where(["map_id IN (?)", @review_response_map_ids])
-      @reviewers = ResponseMap.select("DISTINCT reviewer_id").where(["reviewed_object_id IN (?) and type = ?", @response_ids, @type])
-    when "TeammateReviewResponseMap"
-      #Example query
-      #SELECT distinct reviewer_id FROM response_maps where type = 'TeammateReviewResponseMap' and reviewed_object_id = 711
-      @reviewers = ResponseMap.select("DISTINCT reviewer_id").where(["reviewed_object_id = ? and type = ?", @id, 'TeammateReviewResponseMap'])
-    when "Calibration"
-      participant = AssignmentParticipant.where(parent_id: params[:id], user_id: session[:user].id).first rescue nil
-      if participant.nil?
-        participant = AssignmentParticipant.create(parent_id: params[:id], user_id: session[:user].id, can_submit: 1, can_review: 1, can_take_quiz: 1, handle: 'handle')
-      end
-      @assignment = Assignment.find(params[:id])
-      @review_questionnaire_ids = ReviewQuestionnaire.select("id")
-      @assignment_questionnaire = AssignmentQuestionnaire.where(["assignment_id = ? and questionnaire_id IN (?)", params[:id], @review_questionnaire_ids]).first
-      @questions = @assignment_questionnaire.questionnaire.questions.select{|q| q.type == 'Criterion' or q.type == 'Scale'}
-      @calibration_response_maps = ReviewResponseMap.where(["reviewed_object_id = ? and calibrate_to = ?", params[:id], 1])
-      @review_response_map_ids = ReviewResponseMap.select('id').where(["reviewed_object_id = ? and calibrate_to = ?", params[:id], 0])
-      @responses = Response.where(["map_id IN (?)", @review_response_map_ids])
+        @assignment = Assignment.find(params[:id])
+        @review_questionnaire_ids = ReviewQuestionnaire.select("id")
+        @assignment_questionnaire = AssignmentQuestionnaire.where(["assignment_id = ? and questionnaire_id IN (?)", params[:id], @review_questionnaire_ids]).first
+        @questions = @assignment_questionnaire.questionnaire.questions.select{|q| q.type == 'Criterion' or q.type == 'Scale'}
+        @calibration_response_maps = ReviewResponseMap.where(["reviewed_object_id = ? and calibrate_to = ?", params[:id], 1])
+        @review_response_map_ids = ReviewResponseMap.select('id').where(["reviewed_object_id = ? and calibrate_to = ?", params[:id], 0])
+        @responses = Response.where(["map_id IN (?)", @review_response_map_ids])
     end
   end
+  
+  private
```

Comment:
Although you have already separated some code from `automatic_review_mapping` method, more refactors are still needed. For example, separating the random number generation code into a new private method.